### PR TITLE
Bugfix/fix cider selector problem

### PR DESCRIPTION
**Pull request by ahungry**
This fixes : 
https://github.com/clojure-emacs/cider/issues/2707
-----------------

Before submitting the PR make sure the following things have been done (and denote this
by checking the relevant checkboxes):

- [ x ] The commits are consistent with our [contribution guidelines](../.github/CONTRIBUTING.md)
- [ - ] You've added tests (if possible) to cover your change(s)
- [ x ] All tests are passing (`make test`)
- [ x ] All code passes the linter (`make lint`) which is based on [`elisp-lint`](https://github.com/gonewest818/elisp-lint) and includes
  - [byte-compilation](https://www.gnu.org/software/emacs/manual/html_node/elisp/Byte-Compilation.html), [`checkdoc`](https://www.gnu.org/software/emacs/manual/html_node/elisp/Tips.html), [check-declare](https://www.gnu.org/software/emacs/manual/html_node/elisp/Declaring-Functions.html), packaging metadata, indentation, and trailing whitespace checks.
- [ x ] You've updated the [changelog](../blob/master/CHANGELOG.md) (if adding/changing user-visible functionality)
- [ x ] You've updated the [user manual](../blob/master/doc) (if adding/changing user-visible functionality)

Thanks!

*If you're just starting out to hack on CIDER you might find this [section of its
manual][1] extremely useful.*

[1]: https://docs.cider.mx/cider/contributing/hacking.html

### CODE REVIEW

File: cider-selector.el [hunk 1]
```diff
@@ -139,7 +145,8 @@ is chosen.  The returned buffer is selected with
 
 (def-cider-selector-method ?r
   "Current REPL buffer."
-  (cider-current-repl))
+  (or (cider-current-repl)
```
Comment (bbatsov) [inline cider-selector.el]: The docstring probably needs to be adjusted, as now this does something different.

**Comment by bbatsov** [comment]
Might also be a good idea to expand a bit the documentation for `cider-selector` in the manual.

File: cider-selector.el [hunk 2]
```diff
@@ -49,17 +49,23 @@ DESCRIPTION is a one-line description of what the key selects.")
 Not meant to be set by users.  It's used internally
 by `cider-selector'.")
 
-(defun cider-selector--recently-visited-buffer (mode)
+(defun cider-selector--recently-visited-buffer (mode &optional consider-visible-p)
   "Return the most recently visited buffer, deriving its `major-mode' from MODE.
-Only considers buffers that are not already visible."
+CONSIDER-VISIBLE-P will allow handling of visible windows as well.
+First pass only considers buffers that are not already visible.
```
Comment (bbatsov) [inline cider-selector.el]: I think we can probably simplify the logic and consider all buffers right away. I've mostly copied this code from SLIME back in the day and I didn't think much about it. Treating differently hidden/visible buffers seems weird to me.

Comment (ahungry) [reply cider-selector.el]: I had thought about that, and on one of the commits actually implemented as such, but that had a side-effect of removing the ability to use cider-selector to flip-flop between the latest 2 buffers of the same type (to change from a clojure-mode buffer to the last visited but not shown clojure-mode buffer - not using the second pass would change this operation into a noop essentially).

File: cider-selector.el (hunk 1 above)
Comment (ahungry) [reply cider-selector.el]: updated

File: cider-selector.el (hunk 2 above)
Comment (bbatsov) [reply cider-selector.el]: Got it. Well, I guess it really depends on what do we consider the purpose of this command to be - for me it was always to jump to a source buffer from some special buffer and vice versa. Of course, there's also the question of how people are actually using something and what do they consider useful. :D 

That's why I was thinking we should extend the documentation of `cider-selector` in general so people would know how it's supposed to be used.

**Pull request merged**